stan/math/prim/scal/fun/hypot.hpp

```diff
@@ -23,8 +21,8 @@ namespace math {
  */
 template <typename T1, typename T2>
 inline return_type_t<T1, T2> hypot(const T1& x, const T2& y) {
-  using std::sqrt;
-  return sqrt(square(x) + square(y));
+  using std::hypot;
+  return hypot(x, y);
```

Comment:
just fyi, `std::hypot` is supposed to be a good bit slower. Though also more precise. Not sure which we'd rather go for in this situation

https://stackoverflow.com/questions/32435796/when-to-use-stdhypotx-y-over-stdsqrtxx-yy

Reply:
Huh, I had no idea. I'll revert it for this pull but I'll look into different constructions of ```hypot```, since I think we could do better.

I found this page very interesting: https://deepai.org/publication/an-improved-algorithm-for-hypot-a-b

But of course they didn't report timings.

Reply:
> Is there a situation where you see enough usage of hypot to be worth an optimization effort?

Mainly because it would be introducing a performance regression (in terms of speed, not precision) if we replaced ```sqrt(square(a) + square(b))``` with ```hypot(a, b)```. I'm not sure whether the speed cost would outweigh the precision benefits or whether we'd just end up having to revert it down the line